examples/openssl_engine_example.cpp

```diff
@@ -0,0 +1,311 @@
+                std::cerr << errstr << std::endl;
+                exit(1);
+            }
+            break;
+        case 'X': {
+            char *name, *val;
+
+            name = optarg;
+            if (!(val = strchr(name, '='))) {
+                std::cerr << "%% Expected -X property=value, not " <<
+                    name << std::endl;
+                exit(1);
+            }
+
+            *val = '\0';
+            val++;
+
+            if (conf->set(name, val, errstr) != RdKafka::Conf::CONF_OK) {
+                std::cerr << errstr << std::endl;
+                exit(1);
+            }
+        }
+        break;
+
+        default:
+            goto usage;
+        }
+    }
+
+    if (brokers.empty() || engine_path.empty() || optind != argc) {
+    usage:
+        std::string features;
+        conf->get("builtin.features", features);
+        fprintf(stderr,
+            "Usage: %s [options] -b <brokers> -p <engine-path> \n"
+            "\n"
+            "OpenSSL engine integration example. This example fetches metadata\n"
+            "over SSL channel with broker, established using OpenSSL engine.\n"
+            "\n"
+            "librdkafka version %s (0x%08x, builtin.features \"%s\")\n"
+            "\n"
+            " Options:\n"
+            "  -b <brokers>              Broker address\n"
+            "  -p <engine-path>          Path to OpenSSL engine\n"
+            "  -i <engine-id>            OpenSSL engine id\n"
+            "  -e <engine-callback-data> OpenSSL engine callback_data\n"
+            "  -c <ca-cert-location>     File path to ca cert\n"
+            "  -d [facs..]               Enable debugging contexts: %s\n"
+            "  -X <prop=name>            Set arbitrary librdkafka configuration"
+            " property\n"
+            "\n",
+            argv[0],
+            RdKafka::version_str().c_str(), RdKafka::version(),
+            features.c_str(),
+            RdKafka::get_debug_contexts().c_str());
+        exit(1);
+    }
+
+    if (conf->set("bootstrap.servers", brokers, errstr) != 
+        RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (conf->set("security.protocol", "ssl", errstr) != RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (conf->set("ssl.engine.location", engine_path, errstr) != 
+        RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (ca_location.length() > 0 && 
+        conf->set("ssl.ca.location", ca_location, errstr) != 
+        RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (engine_id.length() > 0 &&
+        conf->set("ssl.engine.id", engine_id, errstr) != RdKafka::Conf::CONF_OK) {
+        std::cerr << errstr << std::endl;
+        exit(1);
+    }
+
+    if (engine_callback_data.length() > 0 &&
```

Comment:
Add a comment saying that engine_callback_data needs to be persistent and outlive the lifetime of the Kafka client handle.